File: doc/quote-azure-k8-backend/Services/ZenQuotesService.cs
using Microsoft.Extensions.Logging;
using quote_azure_k8_backend.Models;
using quote_azure_k8_backend.Data;
using System.Net.Http.Json;
using System.Text.Json;

namespace quote_azure_k8_backend.Services
{
    public class ZenQuotesService : IZenQuotesService
    {
        private readonly HttpClient _httpClient;
        private readonly ILogger<ZenQuotesService> _logger;
        private readonly IQuoteRepository _quoteRepository;

        public ZenQuotesService(HttpClient httpClient, ILogger<ZenQuotesService> logger, IQuoteRepository quoteRepository)
        {
            _httpClient = httpClient;
            _logger = logger;
            _quoteRepository = quoteRepository;
            _httpClient.BaseAddress = new Uri("https://zenquotes.io/api/");
        }

        public async Task<Quote> GetRandomQuoteAsync()
        {
            try
            {
                var response = await _httpClient.GetFromJsonAsync<ZenQuoteResponse[]>("random");
                if (response == null || response.Length == 0)
                {
                    _logger.LogWarning("No quotes returned from ZenQuotes API");
                    throw new InvalidOperationException("No quotes available from external API");
                }

                var zenQuote = response[0];
                return new Quote
                {
                    QuoteText = zenQuote.q,
                    Author = zenQuote.a,
                    LikeCount = 0,
                    CreatedAt = DateTime.UtcNow,
                    Source = "ZenQuotes"
                };
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "Error fetching random quote from ZenQuotes API");
                throw;
            }
        }

        public async Task<List<Quote>> GetMultipleQuotesAsync()
        {
            try
            {
                var response = await _httpClient.GetFromJsonAsync<ZenQuoteResponse[]>("quotes");
                if (response == null || response.Length == 0)
                {
                    _logger.LogWarning("No quotes returned from ZenQuotes API");
                    return new List<Quote>();
                }

                var quotes = new List<Quote>();
                foreach (var zenQuote in response)
                {
                    quotes.Add(new Quote
                    {
                        QuoteText = zenQuote.q,
                        Author = zenQuote.a,
                        LikeCount = 0,
                        CreatedAt = DateTime.UtcNow,
                        Source = "ZenQuotes"
                    });
                }

                return quotes;
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "Error fetching multiple quotes from ZenQuotes API");
                throw;
            }
        }
    }

    public class ZenQuoteResponse
    {
        public string q { get; set; } = string.Empty; // quote text
        public string a { get; set; } = string.Empty; // author
        public string h { get; set; } = string.Empty; // HTML
    }
}

File: k8-quote-api/src/main/java/com/quote/k8/service/ZenQuotesService.java
package com.quote.k8.service;

import com.quote.k8.model.Quote;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
import org.jboss.logging.Logger;

import java.util.List;

@ApplicationScoped
public class ZenQuotesService {

    private static final Logger LOG = Logger.getLogger(ZenQuotesService.class);

    @Inject
    @org.eclipse.microprofile.rest.client.inject.RestClient
    ZenQuotesClient zenQuotesClient;

    public Quote getRandomQuote() {
        try {
            List<ZenQuoteResponse> responses = zenQuotesClient.getRandom();
            if (responses == null || responses.isEmpty()) {
                throw new IllegalStateException("No quotes returned from ZenQuotes API");
            }
            ZenQuoteResponse response = responses.get(0);
            return mapToQuote(response);
        } catch (Exception e) {
            LOG.error("Error fetching random quote from ZenQuotes", e);
            throw new RuntimeException("Failed to fetch quote from ZenQuotes", e);
        }
    }

    public List<Quote> getMultipleQuotes() {
        try {
            List<ZenQuoteResponse> responses = zenQuotesClient.getQuotes();
            if (responses == null || responses.isEmpty()) {
                LOG.warn("No quotes returned from ZenQuotes API");
                return List.of();
            }
            return responses.stream()
                    .map(this::mapToQuote)
                    .toList();
        } catch (Exception e) {
            LOG.error("Error fetching multiple quotes from ZenQuotes", e);
            throw new RuntimeException("Failed to fetch quotes from ZenQuotes", e);
        }
    }

    private Quote mapToQuote(ZenQuoteResponse response) {
        Quote quote = new Quote();
        quote.quoteText = response.q;
        quote.author = response.a;
        quote.likeCount = 0;
        quote.createdAt = java.time.LocalDateTime.now();
        quote.source = "ZenQuotes";
        return quote;
    }

    @RegisterRestClient(baseUri = "https://zenquotes.io/api")
    public interface ZenQuotesClient {
        @jakarta.ws.rs.GET
        @jakarta.ws.rs.Path("/random")
        @jakarta.ws.rs.Produces(jakarta.ws.rs.core.MediaType.APPLICATION_JSON)
        List<ZenQuoteResponse> getRandom();

        @jakarta.ws.rs.GET
        @jakarta.ws.rs.Path("/quotes")
        @jakarta.ws.rs.Produces(jakarta.ws.rs.core.MediaType.APPLICATION_JSON)
        List<ZenQuoteResponse> getQuotes();
    }

    public static class ZenQuoteResponse {
        public String q;  // quote text
        public String a;  // author
        public String h;  // HTML
    }
}
